demos/device_shadow_for_aws/shadow_demo_main.c

```diff
@@ -273,6 +313,127 @@ static void prvUpdateDeltaHandler( MQTTPublishInfo_t * pxPublishInfo );
  */
 static void prvUpdateAcceptedHandler( MQTTPublishInfo_t * pxPublishInfo );
 
+
+/**
+ * @brief Process payload from `/delete/rejected` topic.
+ *
+ * This handler examines the rejected message to look for the reject reason code.
+ * If the reject reason code is `404`, an attempt was made to delete a shadow
+ * document which was not present yet. This is considered to be success for this
+ * demo application.
+ *
+ * @param[in] pxPublishInfo Deserialized publish info pointer for the incoming
+ * packet.
+ */
+static void prvDeleteRejectedHandler( MQTTPublishInfo_t * pxPublishInfo );
+
+/**
+ * @brief Helper function to wait for a response for Shadow delete operation.
+ *
+ * @param[in] pxMQTTContext MQTT context pointer.
+ *
+ * @return pdPASS if successfully received a response for Shadow delete
+ * operation; pdFAIL otherwise.
+ */
+static BaseType_t prvWaitForDeleteResponse( MQTTContext_t * pxMQTTContext );
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvWaitForDeleteResponse( MQTTContext_t * pxMQTTContext )
+{
+    uint8_t ucCount = 0U;
+    MQTTStatus_t xMQTTStatus = MQTTSuccess;
+    BaseType_t xReturnStatus = pdPASS;
```

Comment:
nit: in general it's better to initialize return values to failure rather than success.  No action required.